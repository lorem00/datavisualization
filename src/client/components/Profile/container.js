import React from 'react';
import Profile from './profile';
import './profile.scss';

export default () => (
    <div>
        <div className="page-content">
            <Profile />
        </div>
    </div>
);
