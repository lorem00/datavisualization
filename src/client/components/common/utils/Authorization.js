import React from 'react';
import NotAuthorized from './../../NotAuthorized/NotAuthorized';
import Login from './../../Auth/LoginPage';
import {Redirect, Route, Switch} from 'react-router-dom';

const Authorization = function(WrappedComponent, allowedRoles)
{
    return class WithAuthorization extends React.Component {
        constructor(props) {
            super(props)
        }

        render() {
            const user =  window.sessionStorage.getItem('user');
            const userObj = JSON.parse(user);
            if (userObj && allowedRoles==undefined){
                return <WrappedComponent {...this.props} />
            } else if (userObj && allowedRoles.some(r=> userObj.roles.indexOf(r) >= 0)) {
                return <WrappedComponent {...this.props} />
            } else if (userObj === null) {
                window.location = '/login';
            } else {
                return <NotAuthorized />;
            }
        }
    }
};
export default Authorization;
