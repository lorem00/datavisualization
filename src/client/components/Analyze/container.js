import React from 'react';
import { connect } from 'react-redux';
import Authorization from '../../components/common/utils/Authorization';

class Analyze extends React.Component {
    static defaultProps = {
        serversLastUpdate: null,
    };

    componentDidMount() {

    }

    render() {
        return (
            <div>
                <PageHeading title="Analyze" url="/analyse" />
                <div className="page-content" />
            </div>
        );
    }
}

function mapStateToProps() {
    return {

    };
}

function mapDispatchToProps() {
    return {

    };
}

export default connect(mapStateToProps, mapDispatchToProps)(Authorization(Analyze, ['CUSTOMER']));
