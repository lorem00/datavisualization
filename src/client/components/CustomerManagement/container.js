import React from 'react';
import {BootstrapTable, TableHeaderColumn} from 'react-bootstrap-table';
import Constants from './../common/utils/Constants';
import Authorization from '../../components/common/utils/Authorization';
import {connect} from 'react-redux';

class CustomerManagement extends React.Component {

    renderShowsTotal(start, to, total) {
        return (
            <p style={ { color: 'blue' } }>
                From { start } to { to }, totals is { total }&nbsp;&nbsp;(its a customize text)
            </p>
        );
    }
    editUser(id, field, value) {
        // alert(id + " " + field + " " + value);
        let actualvalue = '';
        let actualfield = '';
        switch(field){
            case 'roledisplay':
                actualfield = 'role';
                actualvalue = Constants.ROLES.find(role => role.display == value).id;
                break;
            default:
                actualfield = field;
                actualvalue = value;
        }
        //this.props.dispatch(saveUserField(id, actualfield, actualvalue));
    }

    format(users) {
        for(let k in users){
            if(users[k].role !=undefined) {
                users[k]['roledisplay'] = users[k].role.display;
            }
        }
        return users;
    }
    onAfterSaveCell(row, cellName, cellValue) {
        //alert(`Save cell ${cellName} with value ${cellValue}`);

        let rowStr = '';
        for (const prop in row) {
            rowStr += prop + ': ' + row[prop] + '\n';
        }
        this.editUser(row.id, cellName, cellValue);
        //alert('Thw whole row :\n' + rowStr);
    }
    onRowSelect(row, isSelected, e) {
        let selected = new Selected();
        selected.rows = [row];
        selected.selected = isSelected;
        this.props.onRowSelect(selected);
    }
    onSelectAll(isSelected, rows) {
        let selected = new Selected();
        selected.rows = rows;
        selected.selected = isSelected;
        this.props.onRowSelect(selected);
    }
    emailFormatter(cell, row) {
        return cell;
        // return (<div>
        //     <Suggestion
        //         id={Math.random()}
        //         type="mitigation"
        //         value="abc"
        //         suggestions={[{name: "a", value: "a"},{name: "aa", value: "aa"},{name: "aaa", value: "aaa"},{name: "aaaa", value: "aaaa"},{name: "aaaaa", value: "aaaaa"}]} />
        // </div>)
    }
    render() {
        let users = this.format([{"lastUpdated":"Apr 25, 2017 9:35:45 PM","password":"","role":{"lastUpdated":"Apr 25, 2017 9:35:44 PM","display":"Administrator","name":"ADMINISTRATOR","id":1},"name":"Ali Jalbani","id":1,"email":"ajalbani@wootcloud.com"},{"lastUpdated":"May 2, 2017 9:04:56 PM","role":{"lastUpdated":"Apr 25, 2017 9:35:44 PM","display":"Administrator","name":"ADMINISTRATOR","id":1},"name":"Srinivas Akella","id":2,"email":"sakella@wootcloud.com"},{"role":{"lastUpdated":"Apr 25, 2017 9:35:44 PM","display":"Administrator","name":"ADMINISTRATOR","id":1},"name":"Siva Mandalam","id":23,"email":"smandalam@wootcloud.com"}]);
        let roleTypes = [];
        for(let role of Constants.ROLES){
            roleTypes.push(role.display)
        }
        const selectRowProp = {
            mode: 'checkbox',
            clickToSelect: true,
            unselectable: [ 1 ],
            bgColor: '#fff2fc',
            // onSelect: this.onRowSelect.bind(this),
            // onSelectAll: this.onSelectAll.bind(this)
        };
        const cellEditProp = {
            mode: 'click',
            blurToSave: true,
            afterSaveCell: this.onAfterSaveCell.bind(this)
        };
        return(
            <div>
                <div className="header-container">
                  <PageHeading title="Customer Management" url="/customers" />
                </div>
                <div className="page-content">
                </div>
            </div>
        )
    }
}

class Selected{
    get rows() {
        return this._rows;
    }

    set rows(value) {
        this._rows = value;
    }

    get selected() {
        return this._selected;
    }

    set selected(value) {
        this._selected = value;
    }
    constructor(){
        this._rows = new Array();
        this._selected = new Boolean();
    }

}

function mapStateToProps(state) {
    return {

    };
}

function mapDispatchToProps(dispatch) {
    return {

    };
}

export default connect(mapStateToProps, mapDispatchToProps)(Authorization(CustomerManagement, ['ADMINISTRATOR']));
