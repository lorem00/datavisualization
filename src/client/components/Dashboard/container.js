import React, { Component } from 'react';
import { connect } from 'react-redux';
import Authorization from '../../components/common/utils/Authorization';

export class Dashboard extends Component {
    static defaultProps = {
        serversLastUpdate: null,
    };

    componentDidMount() {

    }

    render() {
        return (
            <div>
                <div className="page-content" />
            </div>
        );
    }
}

function mapStateToProps() {
    return {

    };
}

function mapDispatchToProps() {
    return {

    };
}

export default connect(mapStateToProps, mapDispatchToProps)(Authorization(Dashboard, ['CUSTOMER']));
