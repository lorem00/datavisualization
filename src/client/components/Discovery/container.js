import React, { Component } from 'react';
import { Nav, NavItem } from 'react-bootstrap';
import { LinkContainer } from 'react-router-bootstrap';
import { connect } from 'react-redux';
import { Redirect, Route, Switch } from 'react-router-dom';
import Authorization from '../../components/common/utils/Authorization';
import GlobalFilters from './GlobalFilters/GlobalFilters';
import DeviceSummary from './DeviceSummary/DeviceSummary';
import DeviceSpectrum from './DeviceSpectrum/DeviceSpectrum';
import DeviceTypes from './DeviceTypes/DeviceTypes';
import DeviceOwnership from './DeviceOwnership/DeviceOwnership';
import DeviceFunctions from './DeviceFunctions/DeviceFunctions';
import DeviceLocations from './DeviceLocations/DeviceLocations';
import DeviceNetworks from './DeviceNetworks/DeviceNetworks';


class Discovery extends Component {
    static defaultProps = {
        serversLastUpdate: null,
    };

    constructor(props) {
        super(props);
        this.state = {
            interval: 'weekly',
            starttime: '',
            endtime: '',
            filter: 'ALL',
        };
    }


    componentDidMount() {

    }

    render() {
        return (
            <div id="discovery">
                <div className="page-content">
                    <Nav bsStyle="tabs" className="container-fluid nav-hor-container">
                        <LinkContainer to="/discovery/spectrum" className="">
                            <NavItem>Tab 1</NavItem>
                        </LinkContainer>
                        <LinkContainer to="/discovery/types" className="horbar-brand">
                            <NavItem>Tab 2</NavItem>
                        </LinkContainer>
                        <LinkContainer to="/discovery/ownership" className="horbar-brand">
                            <NavItem>Tab 3</NavItem>
                        </LinkContainer>
                        <LinkContainer to="/discovery/functions" className="horbar-brand">
                            <NavItem>Tab 4</NavItem>
                        </LinkContainer>
                        <LinkContainer to="/discovery/networks" className="horbar-brand">
                            <NavItem>Tab 5</NavItem>
                        </LinkContainer>
                        <LinkContainer to="/discovery/location" className="horbar-brand">
                            <NavItem>Tab 6</NavItem>
                        </LinkContainer>
                    </Nav>
                    <Switch>
                        <Route path="/discovery/spectrum" component={DeviceSpectrum} />
                        <Route path="/discovery/types" component={DeviceTypes} />
                        <Route path="/discovery/ownership" component={DeviceOwnership} />
                        <Route path="/discovery/functions" component={DeviceFunctions} />
                        <Route path="/discovery/networks" component={DeviceNetworks} />
                        <Route path="/discovery/location" component={DeviceLocations} />
                        <Redirect from="/discovery" to="/discovery/spectrum" />
                    </Switch>
                </div>
            </div>
        );
    }
}

function mapStateToProps() {
    return {

    };
}

function mapDispatchToProps() {
    return {

    };
}

export default connect(mapStateToProps, mapDispatchToProps)(Authorization(Discovery, ['CUSTOMER']));
