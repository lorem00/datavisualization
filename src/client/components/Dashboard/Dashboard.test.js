import React from 'react'
import { shallow, mount } from 'enzyme'
import ShallowRenderer from 'react-test-renderer/shallow'

import {Dashboard} from './container';
import PageHeading from './../common/PageHeading';
import Authorization from '../../components/common/utils/Authorization';

describe('DASHBOARD --- Snapshot', () => {
  it('captures a snapshot of the component', () => {
    const renderedValue =  shallow(<Dashboard />);
    expect(renderedValue).toMatchSnapshot();
  });
});

describe('DASHBOARD', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Dashboard />);
  });
  it('renders the stateless component',  () => {
    expect(wrapper).toBeTruthy();
  });
  it('has a <PageHeading> component', () => {
    expect(wrapper.find(PageHeading)).toHaveLength(1);
  });
  it('has a <div> with className="page-content"', () => {
    expect(wrapper.find('div.page-content')).toHaveLength(1);
  });
});
