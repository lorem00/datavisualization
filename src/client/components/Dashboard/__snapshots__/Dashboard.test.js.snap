// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DASHBOARD --- Snapshot captures a snapshot of the component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <div
      className="header-container"
    >
      <PageHeading
        title="Dashboard"
        url="dashboard"
      />
    </div>
    <div
      className="page-content"
    />
  </div>,
  "nodes": Array [
    <div>
      <div
        className="header-container"
      >
        <PageHeading
          title="Dashboard"
          url="dashboard"
        />
      </div>
      <div
        className="page-content"
      />
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Dashboard
        serversLastUpdate={null}
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Dashboard {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "serversLastUpdate": null,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <div
            className="header-container"
          >
            <PageHeading
              title="Dashboard"
              url="dashboard"
            />
          </div>
          <div
            className="page-content"
          />
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <div
            className="header-container"
          >
            <PageHeading
              title="Dashboard"
              url="dashboard"
            />
          </div>
          <div
            className="page-content"
          />
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Dashboard
    serversLastUpdate={null}
  />,
}
`;
